release/release.go

```diff
@@ -1108,17 +1113,42 @@ func uploadRelease(v version.Version) {
 	}
 }
 
-var linuxRepoVersionsStable = map[string]string{
-	"development": "4.0.0-15-gabcde123", // any non-rc pre-release version will send the package to the "development" repo
-	"testing":     "4.0.0-rc0",          // any rc version will send the package to the "testing" repo
-	"4.3":         "4.3.0",              // any 4.3 stable release version will send the package to the "4.3" repo
-	"4.4":         "4.4.0",              // any 4.4 stable release version will send the package to the "4.4" repo
-	"4.9":         "4.9.0",              // any 4.9 stable release version will send the package to the "4.9" repo
-	"5.0":         "5.0.0",              // any 5.0 stable release version will send the package to the "5.0" repo
+type VersionWithSigningInfo struct {
+	mongoVersionNumber string
+	notaryKeyName string
+	notaryToken string
 }
 
-var linuxRepoVersionsUnstable = map[string]string{
-	"development": "4.0.0-15-gabcde123", // any non-rc pre-release version will send the package to the "development" repo
+var linuxRepoVersionsStable = map[string]VersionWithSigningInfo{
+	"development": {"4.0.0-15-gabcde123", "", ""}, // any non-rc pre-release version will send the package to the "development" repo
+	"testing":     {"4.0.0-rc0", "", ""},          // any rc version will send the package to the "testing" repo
+	"4.3":         {"4.3.0", "server-4.4", os.Getenv("NOTARY_TOKEN_4_4")},              // any 4.3 stable release version will send the package to the "4.3" repo
+	"4.4":         {"4.4.0", "server-4.4", os.Getenv("NOTARY_TOKEN_4_4")},              // any 4.4 stable release version will send the package to the "4.4" repo
+	"4.9":         {"4.9.0", "server-5.0", os.Getenv("NOTARY_TOKEN_5_0")},              // any 4.9 stable release version will send the package to the "4.9" repo
+	"5.0":         {"5.0.0", "server-5.0", os.Getenv("NOTARY_TOKEN_5_0")},              // any 5.0 stable release version will send the package to the "5.0" repo
+}
+
+var linuxRepoVersionsUnstable = map[string]VersionWithSigningInfo{
+	"development": {"4.0.0-15-gabcde123", "", ""}, // any non-rc pre-release version will send the package to the "development" repo
+}
+
+// findArgIndex is the helper function to locate index of provided arg value from an array of arg list
+// The arg list array is assumed to be in such format: ["arg1_name", "arg1_value", "arg2_name", "arg2_value"...]
```

Comment:
I don't think this is a safe assumption. In fact, this assumption is _already_ violated, and the only reason this function works is because our curator invocation happens to have exactly two positional args and no flags that don't take args. 

I'm concerned about this function silently failing in the future if the positional arg count changes or a no-arg flag is added. One possible mitigation is for the caller to panic if this returns `-1`, since we know that the `--api-key` flag should always exist

Reply:
Make sense. I added the panic if `--api-key` not found